src/createClassProxy.js

```diff
@@ -44,17 +44,56 @@ function proxyClass(InitialComponent) {
     return !isEqualDescriptor(staticDescriptors[key], currentDescriptor);
   }
 
+  function instantiate(factory, context, params) {
+    const component = factory();
+
+    try {
+      return component.apply(context, params);
+    } catch (err) {
+      // Native ES6 class instantiation
+      let instance;
+
+      switch (params.length) {
```

Comment:
Just tried that. The same test fails again. Could be because of overriding global.Function. The error is rather cryptic - 'TypeError: Cannot read property 'call' of undefined'. I though CurrentComponent was undefined at that time but it was.